monitor/launch/launcher.go

```diff
@@ -75,42 +75,52 @@ func (nm *NodeMonitor) GetPid() int {
 	return nm.GetProcess().Pid
 }
 
-// Run starts the node monitor.
-func (nm *NodeMonitor) Run(sockPath, bpfRoot string) {
-	nm.SetTarget(targetName)
-	for {
-		os.Remove(sockPath)
-		if err := syscall.Mkfifo(sockPath, 0600); err != nil {
-			log.WithError(err).Fatalf("Unable to create named pipe %s", sockPath)
-			time.Sleep(time.Duration(5) * time.Second)
-		}
+func (nm *NodeMonitor) run(sockPath, bpfRoot string) error {
```

Comment:
Could you comment here that run doesn't return until an error occurs, and that it creates sockPath (the events.sock, I think) and cleans it up on exit. Also, that it exits when it sees an error when reading, including EOF, or when the node-monitor exits (because of `nm.GetProcess()`)